setup_cfg_fmt.py

```diff
@@ -196,6 +196,62 @@ def _python_requires(
         return _format_python_requires(minimum, excluded)
 
 
+def _install_requires(setup_cfg: str) -> List[str]:
+    cfg = configparser.ConfigParser()
+    cfg.read(setup_cfg)
+    raw = cfg.get('options', 'install_requires', fallback='')
+
+    install_requires = raw.strip().splitlines()
+    if not install_requires:
+        return []
+
+    if len(install_requires) == 1:
+        return [_normalize_req(install_requires[0])]
+
+    has_env, has_no_env = [], []
+    for req in install_requires:
+        normalized = _normalize_req(req)
+        if ';' in normalized:
+            has_env.append(normalized)
+        else:
+            has_no_env.append(normalized)
+
+    return [
+        '',
+        *sorted(has_no_env),
+        *sorted(has_env),
+    ]
+
+
+def _normalize_req(req: str) -> str:
+    name, _, envs = req.partition(';')
+    normalized = _normalize_name(name)
+
+    if not envs:
+        return normalized
+
+    return f'{normalized};{envs}'
+
+
+BASE_NAME_REGEX = re.compile(r'[^!=><\s]+')
+REQ_REGEX = re.compile(r'(===|==|!=|~=|>=?|<=?)\s*([^,]+)')
+
+
+def _normalize_name(lib: str) -> str:
```

Comment:
can you demo this function with some unit tests, would be easier to see what's going on here

Reply:
sure